internal/syncd/integrations/ecr/reconciler_test.go

```diff
@@ -0,0 +1,271 @@
+package ecr
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"ship-it/internal"
+
+	"github.com/google/go-github/v26/github"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const inputYaml = `apiVersion: helmreleases.k8s.wattpad.com/v1alpha1
+kind: HelmRelease
+metadata:
+  creationTimestamp: null
+  name: example-microservice
+spec:
+  chart:
+    path: microservice
+    repository: wattpad.s3.amazonaws.com/helm-charts
+    revision: HEAD
+  releaseName: example-release
+  values:
+    image:
+      repository: 723255503624.dkr.ecr.us-east-1.amazonaws.com/bar
+      tag: baz
+status: {}
+`
+
+const expectedYaml = `apiVersion: helmreleases.k8s.wattpad.com/v1alpha1
+kind: HelmRelease
+metadata:
+  creationTimestamp: null
+  name: example-microservice
+spec:
+  chart:
+    path: microservice
+    repository: wattpad.s3.amazonaws.com/helm-charts
+    revision: HEAD
+  releaseName: example-release
+  values:
+    image:
+      repository: 723255503624.dkr.ecr.us-east-1.amazonaws.com/bar
+      tag: 78bc9ccf64eb838c6a0e0492ded722274925e2bd
+status: {}
+`
+
+type MockRepoService struct {
+	mock.Mock
+}
+
+func (m *MockRepoService) GetFile(branch string, path string) (string, error) {
+	args := m.Called(branch, path)
+	return args.String(0), args.Error(1)
+}
+
+func (m *MockRepoService) UpdateFile(msg string, branch string, path string, fileContent []byte) (*github.RepositoryContentResponse, error) {
+	args := m.Called(msg, branch, path, fileContent)
+	return args.Get(0).(*github.RepositoryContentResponse), args.Error(1)
+}
+
+type MockIndexerService struct {
+	mock.Mock
+}
+
+func (m *MockIndexerService) Lookup(repo string) ([]types.NamespacedName, error) {
+	args := m.Called(repo)
+	return args.Get(0).([]types.NamespacedName), args.Error(1)
+}
+
+func TestReconcile(t *testing.T) {
```

Comment:
There is a lot of code repetition throughout these test cases. I tried using named sub-tests to avoid repeatedly mocking out the same dependencies. However, it seems when I call `mock.On()` twice in the same function, only the first call takes effect instead of the second one taking over in the following named sub test. As a result, I must place the cases in different functions entirely. Another work around for this would be to declare the common dependencies as global variables but I am unsure if this is considered a good practice.